### ISSUE TITLE
Scala 2.13: String.collect does not return to String

### ISSUE THREAD

**Issue by Sciss** [scala/bug#11549]
The following works in Scala 2.11 and 2.12:
```scala
def clean(n0: String): String = n0.collect {
  case c if c.isLetterOrDigit   => c
  case c if c >= ' ' && c < '.' => c
  case '.' => '-'
}

assert(clean("__Hello. World!__") == "Hello- World!")
```
but fails in 2.13.0-RC3, because we don't seem to be coming implicitly back from `IndexedSeq[Char]` to `String`. Intentional?

**Comment by dwijnand** [comment]
Looks like `collect` was missed on `StringOps`?

**Comment by Jasper-M** [comment]
RC4 incoming?
Also why doesn't `WrappedString` have specialized versions of `collect`, `map`, `flatMap`? Does `WrappedString` even need to exist?

**Comment by lrytz** [comment]
`StirngOps` add extension methods to strings. The goal of `WrappedString` is to treat a string as a collection, and keep it as such until calling `toString` (`WrappedString.filter` returns a `WrappedString`).

`StringOps.collect` was probably forgotten, so the conversion to `WrappedString` kicks in. `map`, `flatMap`, `collect` overloads in `WrappedString` could be added (https://docs.scala-lang.org/overviews/core/custom-collections.html#appendix-methods-to-overload-to-support-the-same-result-type-principle).

**Comment by hrhino** [comment]
this was fixed by scala/scala#8113, right?
